src/cmd/add.go

```diff
@@ -26,14 +26,20 @@ var addCmd = &cobra.Command{
 		if err != nil {
 			panic(err)
 		}
-		if !template.HasTemplatesDir(appDir) {
-			fmt.Println("no templates found\n\nPlease add templates to the \".exosphere\" folder of your code base.")
-			os.Exit(1)
+		hasTemplates, err := template.HasTemplatesDir(appDir)
+		if err != nil {
+			panic(err)
```

Comment:
Actually. I'll still debate this. These panics are not for user errors. They are for fully unexpected errors such as the filesystem failures. I personally like the idea of having a stacktrace in the rare case that a user ever encounters them as having the stacktrace gives use a much better chance of being able to fix the problem. The `exitWithNoTemplates` func is an example of a user error to me (and one where we could use `log.Fatal`)